### PR TITLE
Redesign tip panel

### LINKED ISSUE #363: design "tip panel" content calculation and presentation
It will be some combination of: 
level: error, warning
and place of an error: process properties, global, node, node property, edge

### PR DESCRIPTION
**Pull request by PPiedel**
This resolves #363

### CODE REVIEW

**Comment by fi-** [comment]
2019-11-22 13_31_13-Nussknacker

1. Errors should wrap, we cant have horizontal scroll here
2. Margin between errors and warnings can be smaller. In `.node-error-section` we have 10px margin bottom. 5px would be enough
3. Disabled warning should look like this: nodeName is disabled
4. if there are multiple nodes disabled why don't we aggregate them to: `nodeName1, nodeName2, nodaname3 are disabled`

File: ui/client/components/Tips.js [hunk 1]
```diff
@@ -53,26 +43,116 @@ export class Tips extends Component {
     }
   }
 
-  printError = (error, suffix, idx) => {
-    return (
-      <div key={idx + suffix} title={error.description}>
-      {(suffix ? suffix + ": " : '') + error.message + (error.fieldName ? `(${error.fieldName})` : "")}
+  errors = (errors) =>
+    <div key={uuid4()} className={"error-tips"}>
+      {this.headerIcon(errors)}
```
Comment (lciolecki) [inline ui/client/components/Tips.js]: What do you think about creating two components like: GlobalErrors and GlobalWarning, where we can move all logic with parsing and presenting data? Now we have all these in one component - I think it's too large :)

**Comment by lciolecki** [comment]
I think we should present warnings and erros in the same way.. Now we present "Errors: node", but at warnings we don't show information "Warnings".

File: ui/client/components/Tips.js [hunk 2]
```diff
@@ -18,29 +19,18 @@ export class Tips extends Component {
     testing: PropTypes.bool.isRequired
   }
 
-  constructor(props) { super(props) }
-
-  tipText = () => {
+  tipsText = () => {
     if (ProcessUtils.hasNoErrorsNorWarnings(this.props.currentProcess)) {
```
[the remark this answers is not in the document]
Comment (PPiedel) [reply ui/client/components/Tips.js]: no … nor is grammatically correct but maybe let's switch to neither … nor ?

[the remark this answers is not in the document]
Comment (lciolecki) [reply ui/client/components/Tips.js]: 'neither' seems to be good ;)

File: ui/client/components/Tips.js (hunk 1 above)
Comment (PPiedel) [reply ui/client/components/Tips.js]: Done

**Comment by PPiedel** [comment]
> I think we should present warnings and erros in the same way.. Now we present "Errors: node", but at warnings we don't show information "Warnings".

We do not merge all warnings into one line, like we do with errors

**Comment by PPiedel** [comment]
> 2019-11-22 13_38_35-Nussknacker
> 
> 1. Add white-space: nowrap to `node-error-tips` to prevent wrapping the "errors in"
> 2. Errors should go to next line if they cant fit horizontally, like inline text. we cant have horizontal scroll here
> 3. Margin between errors and warnings can be smaller. In `.node-error-section` we have 10px margin bottom. 5px would be enough
> 4. Disabled warning should look like this: `nodeName is disabled`
> 5. if there are multiple nodes disabled why don't we aggregate them to: `nodeName1, nodeName2, nodaname3 are disabled`

Done, except 2

**Pull request merged**